excelutils/excel_sbol_utils/library3.py

```diff
@@ -68,6 +92,9 @@ def subcomponents(rowobj): #UPDATE TO WORK WITH CELL DICT, ALLOW CONSTRAINTS
 			template.features.append(sub_part)
 		# template.assemblePrimaryStructure(comp_list)
 		# template.compile(assembly_method=None)
+		if constraints:
+			for constraint in c_list:
+				template.constraints.append(constraint)
```

Comment:
The one thing I don't see is how you match part number to part name. You are expecting part 1 is same as part 2 but whilst the columns are named part 1 etc the parts themselves are not. I think this may require some consideration

Reply:
I think there might still be an issue. I think you need to refer to the subcomponent names/variable component names that are created as part of the function rather than the actual list of parts

Reply:
So, for example, if you wanted to tie two parts together, you're saying that someone may type in the actual part names into the box for constraints rather than just the 'Part 1', etc. generic stuff?

Reply:
No I think saying part 1 is same as part 2 is fine. On the sbol side though it can't be GFP,RFP is the same as GFP, RFP but variable component 1 is same as variable component 2 or something similar.

Reply:
Ok, I think I see what you're saying. Would you want those variable components to be referenced to their objects, then?

Reply:
Yes, via their uri preferably